datasets/blbooksgenre/blbooksgenre.py

```diff
@@ -0,0 +1,380 @@
+                This includes duplication from different annotators""",
+        ),
+        datasets.BuilderConfig(
+            name="raw",
+            version=VERSION,
+            description="""\
+            This version of the dataset includes all the fields from the original dataset including rows without annotation.
+            It includes duplications from different annotators""",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "title_genre_classifiction"
+
+    def _info(self):
+        if self.config.name == "title_genre_classifiction":
+            features = datasets.Features(
+                {
+                    "BL record ID": datasets.Value("string"),
+                    "title": datasets.Value("string"),
+                    "label": datasets.features.ClassLabel(
+                        names=["Fiction", "Non-fiction"]
+                    ),
+                }
+            )
+        if self.config.name == "annotated_raw":
+            features = annotated_raw_features
+        if self.config.name == "raw":
+            features = raw_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        data_file = dl_manager.download(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": data_file,
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _parse_language(self, row: Dict) -> List[str]:
+        languages = row["Languages"]
+        if not languages:
+            return []
+        return languages.split(";")
+
+    def _parse_country(self, row: Dict) -> List[str]:
+        return (
+            row["Country of publication"].split(";")
+            if row["Country of publication"]
+            else []
+        )
+
+    def _parse_place_of_publication(self, row: Dict) -> List[str]:
+        return (
+            row["Place of publication"].split(";")
+            if row["Place of publication"]
+            else []
+        )
+
+    def _parse_all_names(self, row: Dict) -> List[str]:
+        return row["All names"].split(";") if row["All names"] else []
+
+    def _generate_examples(self, filepath, split):
+        """Yields examples as (key, example) tuples."""
+        with open(filepath, encoding="utf-8") as f:
+            reader = csv.DictReader(f)
+            if self.config.name == "title_genre_classifiction":
+                unique = set()
+                id_ = 0
+                for row in reader:
+                    annotated = eval(row["annotated"])
```

Comment:
Using `eval` is not considered safe since it allows code execution. Maybe you can use `ast.literal_eval` instead ? It raises an exception if the input isn't a valid Python datatype

Reply:
I've replaced this with `ast.literal_eval` and should probably fix this in a few other places too 😬